torch/lib/TH/generic/THTensorCopy.c

```diff
@@ -69,20 +78,59 @@ void THTensor_(copyTranspose)(THTensor *tensor, THTensor *src) {
 void THTensor_(copy)(THTensor *tensor, THTensor *src)
 {
   if (tensor == src) return;
-  if (THTensor_(isContiguous)(tensor) && THTensor_(isContiguous)(src) && THTensor_(nElement)(tensor) == THTensor_(nElement)(src)) {
-    real *sp = THTensor_(data)(src);
-    real *rp = THTensor_(data)(tensor);
-    ptrdiff_t sz = THTensor_(nElement)(tensor);
+  ptrdiff_t tensorSize = THTensor_(nElement)(tensor);
+  ptrdiff_t srcSize = THTensor_(nElement)(src);
+  int tensorContig = THTensor_(isContiguous)(tensor);
+  int srcContig = THTensor_(isContiguous)(src);
+
+  int serial_path = 0;
+  int inOMP = omp_in_parallel();
```

Comment:
Shouldn't this be protected with `#ifdef _OPENMP` ?

Reply:
omp_in_parallel is to used to check the code is within openmp region.  If the code is the omp region, there is no need to parallel here. The performance will usually drop if sticks to parallel within omg region.

Reply:
omp_in_parallel is to used to check the code is within openmp region. If the code is the openmp region, there is no need to parallel here. The performance will usually drop if sticks to parallel within openmp region.

Reply:
Yes, but you should ideally add a `#ifdef _OPENMP` as you have done in other parts of the code, as in https://github.com/pytorch/pytorch/pull/2764/files/d04868fc4aa3f0ff2530830c9ce3356b67b58318#diff-5fe5e0b57b11bc4d34842c432d2bdc35R96 for example

Reply:
In particular, will this function exist on system that do not have openmp ?

Reply:
#ifdef _OPENMP is a compiling macro.  It is used to check the machine support Openmp or not when compiling the code.  omp_in_parallel  is check the code is within openmp region when the code executes.

Reply:
But this code will not compile on machine that do not have openmp because the function `omp_in_parallel` does not exist. Or I am missing something?

Reply:
Oh, I am sorry to misunderstand that.  I will fix it.